jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+
+=== Aren't we missing some details about infrastructrue requirements?
+
+In short, yes, we are. This JEP exists to get the conversation started. Once consensus has been reached, 
+a separate Infrastructure Enhancement Proposal (IEP) will be created to go along with this JEP. It would 
+be premature to describe every detail of implementation before consensus is reached.
+
+== Backwards Compatibility
+
+Autorelease introduces no new risks with regard to backwards compatibility of plugins themselves. 
+However, there will be a requirement for plugin maintainers to use a consistent version numbering 
+scheme. Consistent version numbering will prevent problems with sorting versions. This is discussed 
+elsewhere in this document.
+
+With any plugin upgrade, there are backwards compatibility concerns, and Autorelease is no different 
+in that regard. Without Autorelease, there is still nothing stopping a plugin maintainer from releasing 
+a backwards-breaking change.
+
+Plugin maintainers are also not required to use Autorelease at all. By taking no action, these 
+maintainers will see no change at all to the way they do their Jenkins plugin work.
+
+== Security
+
+Autorelease can help to make Jenkins plugins, and their release processes, more secure in a 
+number of ways. Including, but not limited to:
+
+=== Containment of credentials
+
+By using a single system of record for these builds, a service account, maintained by the 
+JENKINS-CERT team, can be used to access GitHub, deployment to Nexus, and deployment to 
+the update centers. Plugin maintainers need not leave their own credentials on a CI server 
+which they don't own, and permissions already in place in their GitHub repositories provide 
+the required controls over who can merge and release.
+
+=== Automatic enforcement of security best practices
+Autorelease builds will all come from `${trusted-ci-server}`. Rules 
+can be put in place on `${trusted-ci-server}`, which can provide implicit enforcement of the 
+Jenkins infrastructure team's security best practices. Compliance to these best practices becomes 
+something that plugin maintainers need not worry about.
+
+== Infrastructure Requirements
+
+We will need a number of things to get this going. The low level technical details will be described 
+in a separate Infrastructure Enhancement Request, so, this should be considered a summary for now:
+
+1. The webhook, configured on each participating plugin . Security implications of this are a bit beyond the scope of this document so far.
+2. A receiver for the aforementioned webhook, because the `${trusted-ci-server}` will be protected 
+behind a VPN
+3. A trusted Jenkins server for performing builds and deployments. 
+4. `buildPlugin` will need code added to validate `Autorelease.yaml` for correctness, and 
+build the plugin according to the settings described therein
+5. Service account(s), managed by the Jenkins Infrastructure team, which provide secure credentials 
```

Comment:
FWIW in JEP-305 the deployment is done outside Jenkins, so while the team may manage this service account it is not at risk of compromise from Jenkins itself.

Reply:
This Jenkins could be on a nonpublic network and report results back using GH statuses. The challenge for incrementals was the deployment of the same artifacts as on public PR build, this doesn't need to have this limitation.